### PR TITLE
ADCM-2852 add logic for apply\revert host-component map for upgrade

### PR DESCRIPTION
**Pull request by SofiaTal**

### CODE REVIEW

File: python/job_runner.py [hunk 1]
```diff
@@ -147,13 +147,34 @@ def run_ansible(job_id):
     sys.exit(ret)
 
 
+def switch_hc(task, action):
+    cluster = task.task_object
+    old_hc = cm.api.get_hc(cluster)
+    new_hc = [*task.post_upgrade_hc_map, *old_hc]
+    task.hostcomponentmap = old_hc
+    task.post_upgrade_hc_map = None
```
Comment (a-alferov) [inline python/job_runner.py:155]: I think need add `task.save()`

File: python/cm/job.py [hunk 2]
```diff
@@ -266,23 +268,48 @@ def add_delta(delta, action, key, fqdn, host):
...
+                            'New components from bundle with upgrade you can only add, not remove',
+                        )
+            if buff == 0:
+                err('INVALID_INPUT', 'hc_acl doesn\'t allow actions with this component')
+            post_upgrade_hc.append(host_comp)
+            new_clear_hc.remove(host_comp)
+
+    old_hc = get_old_hc(api.get_hc(cluster))
+
+    new_hc_list_of_tuple = api.check_hc(cluster, new_clear_hc)
```
Comment (a-alferov) [inline python/cm/job.py:310]: Please think of a better name

**Review by a-alferov — CHANGES_REQUESTED** [review]

**Comment by SofiaTal** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @SofiaTal! Build has been launched!

**Comment by a-alferov** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @a-alferov! Build has been launched!

**Review by a-alferov — APPROVED** [review]

**Pull request merged**